managed/src/test/java/com/yugabyte/yw/controllers/CustomerTaskControllerTest.java

```diff
@@ -265,6 +266,23 @@ public void testTaskHistoryUniverseList() {
     assertAuditEntry(0, customer.uuid);
   }
 
+  @Test
+  public void testTaskHistoryLimit2000() {
+    String authToken = user.createAuthToken();
+    Universe universe1 = createUniverse("Universe 2", customer.getCustomerId());
+    IntStream.range(0, 3000).forEach(i -> createTaskWithStatus(
+        universe.universeUUID, CustomerTask.TargetType.Universe, Create, "Foo", "Running", 50.0));
+    Result result = FakeApiHelper.doRequestWithAuthToken("GET", "/api/customers/" +
+        customer.uuid +  "/universes/" + universe.universeUUID + "/tasks", authToken);
+    assertEquals(OK, result.status());
+    JsonNode json = Json.parse(contentAsString(result));
```

Comment:
nit: Can you parse this to a Map with Json.toJson()?
Same with then parsing values to List.